### PR TITLE
feat(designer): Show loading spinner when connectors API call is still in progress

### LINKED ISSUE #5513: Azure Resource Manager: Failed to see all the actions
### Describe the Bug with repro steps 

1\.Create a Standard Logic app in Brazil South location and add stateful workflow.

2. Add a request trigger. 
3. Add a single-auth Managed Identity Supported Connector (eg. Azure Resource Manager, Key vault, etc.) and choose any operation. 

**Expected:** Should able to see all the actions when we click on see all actions under Azure Resource Manager action.

**Actual:**Failed to see all the actions when we click on see all actions under Azure Resource Manager action.

### What type of Logic App Is this happening in? 

Standard (Portal)

### Which operating system are you using? 

Windows

### Are you using new designer or old designer 

New Designer

### Did you refer to the TSG before filing this issue? [https://aka.ms/lauxtsg](<https://aka.ms/lauxtsg>) 

Yes

### Workflow JSON 

*No response*

### Screenshots or Videos 

[https://github.com/user-attachments/assets/9c87f997-93fb-45fd-b63e-60db9257e782](<https://github.com/user-attachments/assets/9c87f997-93fb-45fd-b63e-60db9257e782>)

[ARM.zip](<https://github.com/user-attachments/files/16756948/ARM.zip>)

### Browser 

Edge

### Additional context 

**Version:** 2\.40826.1.3

### PR DESCRIPTION
**Pull request by ccastrotrejo**
This pull request introduces several changes to enhance the handling of connectors and loading states in the `OperationGroupDetailsPage` and related components. The most important changes include making the `connector` property optional, adding a new `isLoadingConnector` property, and updating the rendering logic to account for the new loading state.

  - Made the `connector` property optional in `OperationGroupDetailsPageProps` and added a new `isLoadingConnector` property.
  - Updated the `OperationGroupDetailsPage` component to use default values for `connector` properties and conditionally render elements based on `isLoadingConnector`.

  - Made the `connector` property optional in `OperationGroupDetailViewProps` and added a new `isLoadingConnector` property.
  - Updated the `OperationGroupDetailView` component to pass the new `isLoadingConnector` property.

  - Added `isLoading` state to the `useAllConnectors` hook and passed it as `isLoadingConnectors`.
  - Updated the `RecommendationPanelContext` component to pass `isLoadingConnectors` to `OperationGroupDetailView`.

Fixes #5513 

#### Video was recorded with simulation of slow 4G connection

https://github.com/user-attachments/assets/4fd2c967-4305-415b-ab60-ee4c99207f94

### CODE REVIEW

File: libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx [hunk 1]
```diff
@@ -226,13 +226,14 @@ export const RecommendationPanelContext = (props: CommonPanelProps) => {
...
           [SELECTION_STATES.CUSTOM_SWAGGER]: selectedOperation ? <CustomSwaggerSelection operation={selectedOperation} /> : null,
-          [SELECTION_STATES.DETAILS]: selectedConnector ? (
+          [SELECTION_STATES.DETAILS]: selectedOperationGroupId ? (
             <OperationGroupDetailView
               connector={selectedConnector}
               groupOperations={allOperationsForGroup}
               filters={filters}
               onOperationClick={onOperationClick}
               isLoading={isLoadingOperations || isLoadingOperationGroup}
+              isLoadingConnector={isLoadingConnectors}
```
Comment (rllyy97) [inline libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx:236]: Won't this not let us look at any connectors if any of them are still loading?

Comment (ccastrotrejo) [reply libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx:236]: Yep, I think thats right. I will change to be loading until the `selectedConnector` exists or is different than undefined. Do you think that will work?

Comment (rllyy97) [reply libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx:236]: Yeah I think that would be better, if the connection is slow and they click on a connector, they may sit there waiting a while until all the connectors load

Comment (ccastrotrejo) [reply libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx]: Yep, you are right I did a couple of tests with sleeps in other connectors types and this was waiting for all of them to be resolved. Thanks for pointing this out

**Review by DanielleCogs — APPROVED** [review]

**Pull request merged**